### PR TITLE
CMDCT-3892: initial app-api service setup with dummy hello world handler

### PR DESCRIPTION
**Pull request by angelaco11**
### Description
Setting up initial app-api service with a dummy hello world handler

---
### How to test
Cloudfront link: https://djc8gs293zsgs.cloudfront.net/
1. Log in as stateuser@test.com 
2. Go to My Account -> Manage Account
3. Look for the text "Hello World!" under the My Account header like so:
Screenshot 2024-09-03 at 12 31 35 PM

### Important updates

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

### CODE REVIEW

File: services/ui-src/src/components/pages/Profile/ProfilePage.tsx [hunk 1]
```diff
@@ -3,10 +3,16 @@ import { Button, Heading, Link, Text } from "@chakra-ui/react";
 import { PageTemplate, Table } from "components";
 import { createEmailLink, useStore } from "utils";
 import verbiage from "verbiage/pages/profile";
+import { useEffect, useState } from "react";
+import { getHelloWorld } from "utils/api/requestMethods/helloWorld";
 
 export const ProfilePage = () => {
   const navigate = useNavigate();
 
+  const [helloWorldMessage, setHelloWorldMessage] = useState<string>(
```
Comment (rocio-desantiago) [inline services/ui-src/src/components/pages/Profile/ProfilePage.tsx:12]: This is the message that's showing. I see that you put //remove 🤔 Should this be the part that gives the Hello World value?

**Review by rocio-desantiago — COMMENTED** [review]
Seeing "default value thing", should "Hello World" be showing instead?

Screenshot 2024-09-04 at 11 43 00 PM

File: services/app-api/.npmignore
Comment (BearHanded) [inline services/app-api/.npmignore:1]: do we actually need a npmignore in app-api? I see all the apps have it but I am uncertain about its use

File: services/app-api/libs/handler-lib.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// utils
```
Comment (BearHanded) [inline services/app-api/libs/handler-lib.ts:1]: please remove these comments

File: services/app-api/utils/authorization.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import { SSMClient, GetParameterCommand } from "@aws-sdk/client-ssm";
+import jwt_decode from "jwt-decode";
+// types
```
Comment (BearHanded) [inline services/app-api/utils/authorization.ts:3]: remove comment

File: services/ui-src/package.json [hunk 4]
```diff
@@ -20,7 +20,7 @@
     "@emotion/styled": "^11.11.5",
     "@hookform/resolvers": "^2.9.11",
     "@vitejs/plugin-react": "^4.3.0",
-    "aws-amplify": "^5.3.19",
+    "aws-amplify": "5.3.4",
```
Comment (BearHanded) [inline services/ui-src/package.json:23]: why are we locking this to a past version in ui-src?

File: services/app-api/utils/authorization.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  allowedRoles: UserRoles[],
+  state?: string
+) => {
+  let isAllowed = false;
+  // decode the idToken
+  if (event?.headers?.["x-api-key"]) {
+    const decoded = jwt_decode(event.headers["x-api-key"]) as DecodedToken;
+    const idmUserRoles = decoded["custom:cms_roles"];
+    const idmUserState = decoded["custom:cms_state"];
+    const mfpUserRole = idmUserRoles
```
Comment (BearHanded) [inline services/app-api/utils/authorization.ts:78]: hcbsUserRole*

**Review by BearHanded — CHANGES_REQUESTED** [review]

File: services/app-api/.npmignore
Comment (angelaco11) [reply services/app-api/.npmignore:1]: i just removed it and it seems fine so let's go with it!

File: services/app-api/libs/handler-lib.ts (hunk 2 above)
Comment (angelaco11) [reply services/app-api/libs/handler-lib.ts]: removed, thank you!

File: services/app-api/utils/authorization.ts (hunk 3 above)
Comment (angelaco11) [reply services/app-api/utils/authorization.ts:3]: removed

File: services/ui-src/package.json (hunk 4 above)
Comment (angelaco11) [reply services/ui-src/package.json]: well i accidentally left the carat off so i added back in, if that was what you were asking 😀

File: services/app-api/utils/authorization.ts (hunk 3 above)
Comment (rocio-desantiago) [inline services/app-api/utils/authorization.ts:3]: did we want to remove `// types` here too?

**Review by rocio-desantiago — DISMISSED** [review]
I'm seeing the expected `Hello World`
Not sure if there's a trailing comment but looks good to me 👍

File: services/app-api/utils/authorization.ts (hunk 3 above)
Comment (angelaco11) [reply services/app-api/utils/authorization.ts]: removed

**Review by benmartin-coforma — DISMISSED** [review]
Solid start 👍

**Review by ailZhou — APPROVED** [review]
lgtm! not my area of expertise but the code did not look weird and i was able to see hello world 🎉

**Review by rocio-desantiago — APPROVED** [review]

**Review by ailZhou — APPROVED** [review]

File: services/ui-src/package.json [hunk 6]
```diff
@@ -20,7 +20,7 @@
     "@emotion/styled": "^11.11.5",
     "@hookform/resolvers": "^2.9.11",
     "@vitejs/plugin-react": "^4.3.0",
-    "aws-amplify": "^5.3.19",
+    "aws-amplify": "^5.3.4",
```
Comment (BearHanded) [inline services/ui-src/package.json:23]: This is still downgrading the minimum version, we probably want to keep the min version at 5.3.19 unless we have a reason to downgrade

**Review by BearHanded — CHANGES_REQUESTED** [review]

File: services/ui-src/package.json (hunk 6 above)
Comment (angelaco11) [reply services/ui-src/package.json:23]: changed this to go back to 5.3.19

**Review by BearHanded — APPROVED** [review]

**Pull request merged**